core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/LandingZoneCreationStatusRunnerSpec.scala

```diff
@@ -0,0 +1,455 @@
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(CreationStatuses.Ready),
+      ArgumentMatchers.eq(None)
+    )
+    verify(billingRepository).updateLandingZoneId(ArgumentMatchers.eq(billingProjectName), ArgumentMatchers.eq(lzId))
+  }
+
+  it should "set an error in the billing project and return complete status when there are errors but no landing zone" in {
+    val userEmail = "user@email.com"
+    val ctx = mock[RawlsRequestContext]
+    val billingProjectName = RawlsBillingProjectName(UUID.randomUUID().toString)
+    val monitorRecord = WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      WorkspaceManagerResourceMonitorRecord.JobType.AzureLandingZoneResult,
+      None,
+      Some(billingProjectName.value),
+      Some(userEmail),
+      new Timestamp(Instant.now().toEpochMilli)
+    )
+    val wsmDao = mock[WorkspaceManagerDAO]
+    val failureMessage = "this is a very specific failure message"
+    val landingZoneResult = new AzureLandingZoneResult()
+      .errorReport(new bio.terra.workspace.model.ErrorReport().message(failureMessage))
+    when(
+      wsmDao.getCreateAzureLandingZoneResult(
+        ArgumentMatchers.eq(monitorRecord.jobControlId.toString),
+        ArgumentMatchers.any()
+      )
+    ).thenReturn(landingZoneResult)
+    val billingRepository = mock[BillingRepository]
+    when(
+      billingRepository.updateCreationStatus(
+        ArgumentMatchers.eq(billingProjectName),
+        ArgumentMatchers.eq(CreationStatuses.Error),
+        ArgumentMatchers.any[Some[String]]()
+      )
+    ).thenAnswer { invocation =>
+      val message: Option[String] = invocation.getArgument(2)
+      assert(message.get.contains(failureMessage))
+      Future.successful(1)
+    }
+    val runner =
+      spy(new LandingZoneCreationStatusRunner(mock[SamDAO], wsmDao, billingRepository, mock[GoogleServicesDAO]))
+    doReturn(Future.successful(ctx)).when(runner).getUserCtx(ArgumentMatchers.eq(userEmail))(ArgumentMatchers.any())
+
+    whenReady(runner.run(monitorRecord))(_ shouldBe WorkspaceManagerResourceMonitorRecord.Complete)
+
+    verify(billingRepository).updateCreationStatus(
+      ArgumentMatchers.eq(billingProjectName),
+      ArgumentMatchers.eq(CreationStatuses.Error),
+      ArgumentMatchers.any[Some[String]]()
+    )
+  }
+
+  it should "set an error in the billing project and return incomplete status when there are no errors or landing zone" in {
+    val userEmail = "user@email.com"
+    val ctx = mock[RawlsRequestContext]
+    val billingProjectName = RawlsBillingProjectName(UUID.randomUUID().toString)
+    val monitorRecord = WorkspaceManagerResourceMonitorRecord(
+      UUID.randomUUID(),
+      WorkspaceManagerResourceMonitorRecord.JobType.AzureLandingZoneResult,
+      None,
+      Some(billingProjectName.value),
+      Some(userEmail),
+      new Timestamp(Instant.now().toEpochMilli)
+    )
```

Comment:
It seems like some of this setup could be moved to a beforeEach or helper method to make the individual test cases easier to read.

Reply:
You're probably correct on that. 
I'm generally averse to setting up mocks like that, since it often degrades into a series of optional args to customize them; but here there's enough repetition to make it worth it